Chapter7/db_revisit.erl

```diff
@@ -0,0 +1,19 @@
+-module(db_revisit).
+-export([new/0, destroy/1, write/2, delete/2, read/2, match/2]).
+-include("db.hrl").
+
+% Exercise 7-3: The db.erl Exercise Revisited
+% Don't know what the book means by "Test your results using the database server developed in Exercise 5-1 in Chapter 5."
+
+new() 				-> [].
+destroy(_) 			-> ok.
+write(Data, Db) 	-> [Data | Db].
+delete(Data, Db) 	-> Db -- match(Data, Db).
+
+read(_, [])										-> {error, key_not_found};
+read(#data{key=Key}, [New = #data{key=Key}|_])	-> {ok, New#data.data};
```

Comment:
I've updated code to take Db as first argument, and I've also cleaned up some of the overuse of "#data"